static/js/src/public/store/packages.js

```diff
@@ -50,7 +59,14 @@ class initPackages {
   }
 
   fetchPackageList() {
-    return fetch("/packages.json").then((result) => result.json());
+    if (this._filters.q) {
+      const queryUrl = this._filters.q.join(",");
+      return fetch(`/packages.json?q=${queryUrl}`).then((result) =>
```

Comment:
In a follow-up can you update any of our fetch usage to handle the failure cases as well? It should probably also have a timeout so that the request doesn't hang for a disproportionate amount of time.

Thx